### PR TITLE
Add initial documentation for testing. r=vporof

### PR DESCRIPTION
**Pull request by jsantell**

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+**Disclaimer**: Test paths that target multiple test suites (`npm test test/**/*.js`) will most likely fail, and the path should only target files within a suite (`npm test test/unit/**/*.js`, or `npm test test/unit`), and if you want to run all tests, simply fire `npm test`. This is due to globbing files and running tests in the appropriate suite.
+
+## Test Types
+
+All tests run in a [babelified](http://babeljs.io) environment, running a [mocha](http://mochajs.org/) suite with [expect](https://www.npmjs.com/package/expect) assertions.
+
+### Unit Tests
+
+Unit tests are the fastest tests to run, and are encouraged over other tests if possible. These tests can directly access source components found in `app/`, and run in a node environment. We run tests on our build system, lint, profile service, and UI components.
```
Comment (victorporof) [inline docs/testing.md]: Not only in `app/`, but also any other directory like `build`.

File: docs/testing.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+#### Testing UI Components
+
+The browser React app can be tested via unit tests. Explore [/test/unit/ui/browser/](/test/unit/ui/browser), where we have action, model, views and widget tests, having a similar directory structure to the app. For the React views, we use [enzyme](https://www.npmjs.com/package/enzyme) to test shallow renderings of the React components.
+
+### Renderer Tests
+
+Renderer tests use [electron-mocha](https://www.npmjs.com/package/electron-mocha) to run tests inside an Electron app's renderer process. Note, this Electron app is not ours, but just a small app that wires up the test communication, and provides our tests with an environment that is the same as what our browser app runs in, provides webviews, and Electron APIs. The browser React app is injected into the DOM before the tests run. Make sure you clean up your tests!
```
Comment (victorporof) [inline docs/testing.md]: Not sure about "components that require the DOM to exist". `enzyme` can render components headlessly without any issues (and I do mean "render", and check DOM data afterwards, not just shallow rendering). We should be careful of our wording here.

**Comment by victorporof** [comment]
r+

File: docs/testing.md (hunk 2 above)
Comment (jsantell) [inline docs/testing.md]: Should clarify with "real" DOM?

Comment (jsantell) [inline docs/testing.md]: Clarified below

**Pull request merged**